quickwit/quickwit-control-plane/src/indexing_scheduler/change_tracker.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2024 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::sync::{Arc, Mutex};
+
+/// This object makes it possible to track for the completion of the next rebuild.
+pub struct RebuildNotifier {
+    generation_processed_tx: Arc<Mutex<tokio::sync::watch::Sender<usize>>>,
```

Comment:
This seems like a [broadcast channel](https://docs.rs/tokio/latest/tokio/sync/broadcast/index.html).

Reply:
Not exactly.
- broadcast don't let me look at the head from the sender
- broadcast channel usually have a capacity (we can set it to one, and older values are then dropped), but then we need to handle the associated error.